ldk/node/src/hostClients/browserClient.test.ts

```diff
@@ -0,0 +1,156 @@
+import { mocked } from 'ts-jest/utils';
+import createMockInstance from 'jest-create-mock-instance';
+import * as Services from '../grpc/browser_grpc_pb';
+import * as Messages from '../grpc/browser_pb';
+import { ConnInfo } from '../grpc/broker_pb';
+import { BrowserClient } from './browserClient';
+import { Session } from '../grpc/session_pb';
+import { Logger } from '../logging';
+import {
+  captureMockArgument,
+  createCallbackHandler,
+  createStreamingHandler,
+  createWaitHandler,
+  defaultConnInfo,
+  defaultSession,
+  identityCallback,
+} from '../jest.helpers';
+import { BrowserSelectedTextResponse } from './browserService';
+
+jest.mock('../grpc/browser_grpc_pb');
+
+const MockClientClass = mocked(Services.BrowserClient);
+
+const logger = new Logger('test-logger');
+
+describe('BrowserClient', () => {
+  let subject: BrowserClient;
+  let mockGRPCClient: jest.Mocked<Services.BrowserClient>;
+  let connInfo: ConnInfo.AsObject;
+  let session: Session.AsObject;
+
+  beforeEach(async () => {
+    jest.resetAllMocks();
+    subject = new BrowserClient();
+    connInfo = defaultConnInfo;
+    session = defaultSession;
+
+    mockGRPCClient = createMockInstance(Services.BrowserClient);
+    mockGRPCClient.waitForReady.mockImplementation(createWaitHandler());
+    MockClientClass.mockImplementation(() => mockGRPCClient as any);
+
+    await expect(
+      subject.connect(connInfo, session, logger),
+    ).resolves.toBeUndefined();
+  });
+
+  describe('#queryActiveURL', () => {
+    let sentRequest: Messages.BrowserActiveURLRequest;
+    let sentResponse: Messages.BrowserActiveURLResponse;
+    let queryResult: Promise<string>;
+
+    beforeEach(async () => {
+      sentResponse = new Messages.BrowserActiveURLResponse();
```

Comment:
I wondered to myself if there's value in setting an explicit value for `Url` property in the response, and then testing that that's returned but we are testing equality on line 65. I think I'm hesitating over this because GRPC primitive properties are auto-set at first.

Yeah this is probably fine :)

Reply:
Oh yeah, I was setting that for some things, but at some point, for expediency, I decided that equality would do for the moment.

Reply:
As in, in the past, sometimes I've even set values to be a `jest.fn()` to get away from magic stringiness in tests (a little extreme, yeah...).

Reply:
I've thought about this some more, and I'll actually set explicit values. I agree that there are likely places where this might pass incorrectly.

Reply:
Addressed